src/Path.php

```diff
@@ -215,7 +215,7 @@ public static function check($path)
 	 */
 	public static function clean($path, $ds = DIRECTORY_SEPARATOR)
 	{
-		if (!\is_string($path))
+		if (!\is_string($path) && !empty($path))
```

Comment:
Should be 
`if (!\is_string($path) || empty($path)) {...}`
to make sense.

On the other hand: Why not 

```php
if (!\is_string($path)) {...}

if (empty(trim($path))) {
    $path = JPATH_ROOT;
}
```
?